### ISSUE TITLE
calling end quickly precludes a write

### ISSUE THREAD

**Issue by chazomaticus** [myndzi/syslog2#4]
Perhaps related to #3, if I call `end` soon after a `write`, the `write` never happens.

```js
var Syslog = require('syslog2');
var log = Syslog.create({
    connection: {
        type: 'udp',
        host: '127.0.0.1',
        port: 514,
    }
}, function () {
    log.write("test", function () {
        log.end();
    });
});
```

Running under `strace` shows the problem:

```
$ strace node test.js 2>&1 | grep '^socket\|^send'
socket(PF_INET, SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, IPPROTO_IP) = 10
$
```

There's no send.  If you wrap the `end` call in a timeout, like this:

```js
var Syslog = require('syslog2');
var log = Syslog.create({
    connection: {
        type: 'udp',
        host: '127.0.0.1',
        port: 514,
    }
}, function () {
    log.write("test", function () {
        setTimeout(function () {
            log.end();
        }, 100);
    });
});
```

You can see the correct behavior:

```
$ strace node test.js 2>&1 | grep '^socket\|^send'
socket(PF_INET, SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, IPPROTO_IP) = 10
sendmsg(10, {msg_name(16)={sa_family=AF_INET, sin_port=htons(514), sin_addr=inet_addr("127.0.0.1")}, msg_iov(1)=[{"<133>1 2015-02-05T03:02:52.00-08"..., 64}], msg_controllen=0, msg_flags=0}, 0) = 64
$
```

Note that changing `setTimeout` to `process.nextTick` does *not* fix the behavior; it seems to need some actual time passage.

**Comment by myndzi** [comment]
...which means it needs to wait on some event, somewhere. I'm hoping this doesn't come down to the native dgram library

**Comment by myndzi** [comment]
I assume the fix for #3 fixes this one as well. Please verify that it behaves as expected as of syslog2?

**Comment by chazomaticus** [comment]
Yes, your patch to udp-streams2 fixed this issue as well.  Thanks for your help and fast turnaround!
